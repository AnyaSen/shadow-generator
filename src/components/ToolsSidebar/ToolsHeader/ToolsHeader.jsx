import React from "react";
import Styles from "./ToolsHeader.module.scss";
import { Link } from "react-router-dom";

import Button from "../../Buttons/Button/Button";

export default function ToolsHeader() {
  return (
    <div className={Styles.ToolsNav}>
      <h2>TOOLS</h2>

      <Link to="/code">
        <Button text="Generate Code" />
      </Link>
    </div>
  );
}
