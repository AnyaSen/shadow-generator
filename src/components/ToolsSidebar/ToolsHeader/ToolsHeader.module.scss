@import "../../../assets/scss/variables";
@import "../../../assets/scss/breakpoints";

.ToolsNav {
  display: flex;
  align-items: center;
  justify-content: space-between;
  position: absolute;
  top: 0;
  z-index: 2;
  background-color: $color-white;
  padding: 0 $space-3;
  box-shadow: $shadow-2;
  width: 45.3%;

  @include lg {
    position: static;
    width: 100%;
  }
}
