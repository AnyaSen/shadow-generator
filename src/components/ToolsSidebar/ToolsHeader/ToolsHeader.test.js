import React from "react";
import { BrowserRouter } from "react-router-dom";
import { render, cleanup } from "@testing-library/react";

import ToolsHeader from "./ToolsHeader";

describe("<ToolsHeader />", () => {
  afterEach(cleanup);

  test("Should have correct text and have 2 children.", () => {
    const { getByText, getByTestId, container } = render(
      <BrowserRouter>
        <ToolsHeader />
      </BrowserRouter>
    );
    const hearderText = "TOOLS";

    const toolsHeader = getByText(hearderText);

    const toolsHeaderContainer = getByTestId("Header");

    expect(toolsHeaderContainer.children.length).toEqual(2);

    expect(toolsHeader).toBeTruthy();

    expect(container.firstChild).toMatchSnapshot();
  });
});
