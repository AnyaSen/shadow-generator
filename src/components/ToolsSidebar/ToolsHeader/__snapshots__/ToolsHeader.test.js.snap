// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ToolsHeader /> Should have correct text and have 2 children. 1`] = `
<div
  class="ToolsNav"
  data-testid="Header"
>
  <h2>
    TOOLS
  </h2>
  <a
    href="/code"
  >
    <button
      class="Button"
    >
      Generate Code
    </button>
  </a>
</div>
`;
